pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -225,6 +225,10 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 				return nil, fmt.Errorf("running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false. /proc/swaps contained: %v", swapLines)
 			}
 		}
+	} else {
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) && !cgroups.IsCgroup2UnifiedMode() {
```

Comment:
I wasn't thinking about it before, but now I realize that the problem with this is that all tests running today with `--fail-on-swap=false` will stop working as the feature gate will be enabled by default.

Even though it is a desirable behavior to fail fast long term, breaking many scenarios is not ideal.

One option is to change the default value for the feature gate.

Reply:
+ @deads2k (PRR reviewer for the SWAP KEP) for the opinion

Reply:
> I wasn't thinking about it before, but now I realize that the problem with this is that all tests running today with --fail-on-swap=false will stop working as the feature gate will be enabled by default.

Yes this is exactly right, and was one of the reasons I asked to do so in a follow-up :)

I think you're right, and that the right approach moving forward is:
1. Introduce a PR that would adapt the tests.
2. Merge this PR after making sure it won't break any lanes.

> One option is to change the default value for the feature gate.

What do you mean by that? Isn't the default behavior for the feature gate is to be disabled?
IIUC all we need to do is remove all jobs that are defined with cgroup v1 + NodeSwap enabled. We should keep (and maybe add more? but outside of this PR's scope) the jobs with swap + v2.

WDYT?